File: weather-app/app/src/main/java/com/rperazzo/weatherapp/utils/WeatherManager.java
package com.rperazzo.weatherapp.utils;

import com.rperazzo.weatherapp.services.WeatherService;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class WeatherManager {

    private static OkHttpClient mClient = new OkHttpClient();

    public static WeatherService getService() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(Const.API_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(mClient)
                .build();

        return retrofit.create(WeatherService.class);
    }




}

File: weather-app/app/src/main/java/com/rperazzo/weatherapp/MVVM/model/service/WeatherManager.kt
package com.rperazzo.weatherapp.MVVM.model.service;

import com.rperazzo.weatherapp.MVVM.utils.Constantes
import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

object WeatherManager {

    private val mClient: OkHttpClient = OkHttpClient();

    private val retrofit: Retrofit = Retrofit.Builder()
            .baseUrl(Constantes.API_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .client(mClient)
            .build();

    fun getService() = retrofit.create(WeatherService::class.java)
}
